platforms/documentation/docs/src/docs/userguide/dep-man/declaring-repositories/supported_metadata_formats.adoc

```diff
@@ -0,0 +1,162 @@
+
+[source,xml]
+----
+<project xmlns="http://maven.apache.org/POM/4.0.0">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.example</groupId>
+  <artifactId>my-library</artifactId>
+  <version>1.0</version>
+</project>
+----
+
+[[sub:supported-md-ivy]]
+=== Ivy files
+
+Gradle also supports http://ant.apache.org/ivy/[Apache Ivy metadata files].
+Gradle will first look for an `ivy.xml` file, but if this file contains a special marker, it will use <<#sub:supported-md-gradle, Gradle Module Metadata>> instead.
+
+[source,xml]
+----
+<ivy-module version="2.0">
+  <info organisation="com.example" module="my-library" revision="1.0"/>
+  <dependencies>
+    <dependency org="org.example" name="dependency" rev="1.2"/>
+  </dependencies>
+</ivy-module>
+----
+
+[[sec:supported_metadata_sources]]
+== Supported metadata sources
+
+When searching for a module in a repository, Gradle checks for supported metadata file formats by default.
+
+In a Maven repository, Gradle looks for `.pom` files.
+In an Ivy repository, it checks for `ivy.xml` files.
+And in a flat directory repository, it looks directly for `.jar` files without expecting any metadata.
+Starting with Gradle 5.0, Gradle also looks for `.module` (Gradle module metadata) files.
+
+If you define a custom repository, you can configure how Gradle searches for metadata.
+For instance, you can set up a Maven repository that only contains JAR files without POM files.
+This is done by configuring metadata sources for the repository:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=maven-repo-with-metadata-sources]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=maven-repo-with-metadata-sources]"]
+====
+
+You can specify multiple metadata sources, and Gradle will search through them in a predefined order.
+The following metadata sources are supported:
+
+[%header%autowidth,compact]
+|===
+| Metadata source | Description | Order | Maven | Ivy / flat dir
+
+| `gradleMetadata()`
+| Look for Gradle `.module` files
+| 1st
+| yes
+| yes
+
+| `mavenPom()`
+| Look for Maven `.pom` files
+| 2nd
+| yes
+| yes
+
+| `ivyDescriptor()`
+| Look for `ivy.xml` files
+| 2nd
+| no
+| yes
+
+| `artifact()`
+| Look directly for artifact
+| 3rd
+| yes
+| yes
+|===
+
+The default behavior for Ivy and Maven repositories has changed with Gradle 6.0, where `artifact()` is no longer included by default, improving efficiency when modules are missing.
+
+To restore the old behavior or configure new behavior in older Gradle versions, you can specify the metadata sources explicitly:
+
+[source,kotlin]
+----
+mavenCentral {
+    metadataSources {
+        mavenPom()
+        artifact()
+    }
+}
+----
+
+Since Gradle 5.3, when parsing metadata files (Ivy or Maven), Gradle checks for a marker that indicates the presence of a matching Gradle Module Metadata file.
```

Comment:
Again, I don't think we really need to have the version numbers in here. This has been the behavior for a long time.

Reply:
Done.